File: php/channel_partner/getChannelPartners.php
<?php

ini_set('display_errors', 1);

/*
    Retrieves a list of all channel partners configured for this merchant.  If the API KEY used is tied to a specific
    Channel Partner, then the results will contain only that Channel Partner.
 */


use ultracart\v2\api\ChannelPartnerApi;

require_once '../vendor/autoload.php';
require_once '../constants.php';


$channel_partner_api = ChannelPartnerApi::usingApiKey(Constants::CHANNEL_PARTNER_API_KEY);
$api_response = $channel_partner_api->getChannelPartners();


if ($api_response->getError() != null) {
    error_log($api_response->getError()->getDeveloperMessage());
    error_log($api_response->getError()->getUserMessage());
    exit();
}

$channel_partners = $api_response->getChannelPartners();

echo '<html lang="en"><body><pre>';
foreach ($channel_partners as $channel_partner) {
    var_dump($channel_partner);
}
echo '</pre></body></html>';

File: java/src/channel_partner/GetChannelPartners.java
package channel_partner;

import com.ultracart.admin.v2.ChannelPartnerApi;
import com.ultracart.admin.v2.models.ChannelPartner;
import com.ultracart.admin.v2.models.ChannelPartnersResponse;

import java.util.List;

public class GetChannelPartners {
  /*
      Retrieves a list of all channel partners configured for this merchant.  If the API KEY used is tied to a specific
      Channel Partner, then the results will contain only that Channel Partner.
   */
  public static void execute() {
    System.out.println("--- GetChannelPartners ---");

    try {
      // Create channel partner API instance using API key
      ChannelPartnerApi channelPartnerApi = new ChannelPartnerApi(common.Constants.CHANNEL_PARTNER_API_KEY);
      ChannelPartnersResponse apiResponse = channelPartnerApi.getChannelPartners();

      if (apiResponse.getError() != null) {
        System.err.println(apiResponse.getError().getDeveloperMessage());
        System.err.println(apiResponse.getError().getUserMessage());
        System.exit(1);
      }

      List<ChannelPartner> channelPartners = apiResponse.getChannelPartners();

      for (ChannelPartner channelPartner : channelPartners) {
        System.out.println(channelPartner);
      }

      System.out.println("Retrieved " + channelPartners.size() + " channel partners");
    } catch (Exception ex) {
      System.out.println("Error: " + ex.getMessage());
      ex.printStackTrace();
    }
  }
}